<launch>
  <arg name="db_path" default="/opt/strands/ros_datacentre"/>
  <arg name="port" default="62345" />
  <arg name="machine" default="localhost" />
  <arg name="user" default="" />

  <param name="datacentre_port" value="$(arg port)" />
  <param name="datacentre_host" value="$(optenv HOSTNAME localhost)" />

  <machine name="$(arg machine)" address="$(arg machine)" env-loader="/opt/strands/strands_catkin_ws/devel/env.sh" user="$(arg user)" default="true"/>
  
  <node name="mongo_server" pkg="ros_datacentre" type="mongodb_server.py" output="screen">
    <param name="database_path" value="$(arg db_path)"/>
  </node>

  <node name="config_manager" pkg="ros_datacentre" type="config_manager.py" output="screen"/>

  <node name="message_store" pkg="ros_datacentre" type="message_store_node.py" output="screen"/>

  <node name="replicator_node" pkg="ros_datacentre" type="replicator_node.py" output="screen"/>

  <!-- rosout and diagnostic topic logger -->
  <!-- <node name="diagnostics_logger" pkg="strands_diagnostics" type="logger" output="screen"/> -->
</launch>
